django/db/models.py

```diff
@@ -466,18 +478,75 @@ def get_snippet_def(cls):
 
 class SmallMoleculeBatch(ReagentBatch):
 
-    chemical_synthesis_reference = _TEXT(**_NULLOKSTR)
-    purity                  = _TEXT(**_NULLOKSTR)
-    purity_method           = _TEXT(**_NULLOKSTR)
-    aqueous_solubility      = models.DecimalField(max_digits=4, decimal_places=2, null=True)
+    _molecular_weight = models.DecimalField(
+        db_column='molecular_weight', max_digits=10, decimal_places=6, null=True) 
+    _molecular_formula = _TEXT(db_column='molecular_formula', **_NULLOKSTR)
+    _chemical_synthesis_reference = _TEXT(**_NULLOKSTR)
+    _purity= _TEXT(**_NULLOKSTR)
+    _purity_method = _TEXT(**_NULLOKSTR)
+    aqueous_solubility = models.DecimalField(
+        max_digits=4, decimal_places=2, null=True)
     aqueous_solubility_unit = models.TextField(null=True)
-    inchi                   = _TEXT(**_NULLOKSTR)
-    inchi_key               = _TEXT(**_NULLOKSTR)
-    smiles                  = _TEXT(**_NULLOKSTR)
+    _inchi = _TEXT(**_NULLOKSTR)
+    _inchi_key = _TEXT(**_NULLOKSTR)
+    _smiles = _TEXT(**_NULLOKSTR)
 
-class Cell(Reagent):
+    def get_salt_id(self):
+        return self.reagent.salt_id  
+      
+    def get_molecular_weight(self, is_authenticated=False):
+        if(not self.reagent.is_restricted or is_authenticated):
```

Comment:
We ought to write a decorator to factor out this restriction pattern... No need to do it for this PR but let's keep it in mind.